File: libraryFinal/src/main/java/com/library/Model/MiniBook.java
package com.library.Model;

public class MiniBook extends Product{

    public MiniBook(){
        super.pc = new PublishingCompany();
        super.author = new Author();
    }

    @Override
    public void setPublishingCompanyName(String pcName){
        pc.setPcName(pcName);
    }
}

File: libraryCSharp/Model/MiniBook.cs
namespace libraryCSharp.Model
{
    public class MiniBook : Product
    {
        public MiniBook()
        {
            base.pc = new PublishingCompany();
            base.author = new Author();
        }
        public override string ToString()
        {
            return base.ToString();
        }
    }
}